staging/src/k8s.io/apiserver/pkg/cel/types.go

```diff
@@ -452,6 +452,10 @@ func (rt *DeclTypeProvider) FindFieldType(typeName, fieldName string) (*ref.Fiel
 	}
 
 	f, found := st.Fields[fieldName]
+	if !found {
+		f, found = st.Fields["__"+fieldName+"__"]
+	}
+
```

Comment:
Let's instead change the code that writes values to Fields to not escape values.  This approach is problematic in the pathological case (use has a field named `if` and a field named `__if__`).

Reply:
The problem is existing without the current PR: https://github.com/kubernetes/kubernetes/issues/126132 so this doesn't introduce new issue :)

Reply:
Can we narrow this down to not-already-escaped names?  I think that's enough to take care of problems related to multiple-levels of escaping.  For example:
```go
if !found && !strings.HasPrefix(fieldName, "__") {
...
}
```